tools/serverpod_cli/lib/src/analyzer/entities/validation/restrictions.dart

```diff
@@ -34,6 +35,23 @@ class Restrictions {
         span,
       ));
     }
+
+    if (_countClassNames(content, protocolEntities) > 1) {
```

Comment:
This is a n-squared operation if we go through all files, right? Maybe we want to refactor in the future to cache data for quicker lookups? (Add a TODO)

Reply:
true, where n is the number of protocol files. Comment added.